src/sidebar/markdown-commands.js

```diff
@@ -10,16 +10,17 @@
 /**
  * Describes the state of a plain text input field.
  *
- * interface EditorState {
- *   text: string;
- *   selectionStart: number;
- *   selectionEnd: number;
- * }
+ * @typedef EditorState
+ * @prop {string} text
+ * @prop {number} selectionStart
+ * @prop {number} selectionEnd
  */
 
 /**
  * Types of Markdown link that can be inserted with
  * convertSelectionToLink()
+ *
+ * @enum {number}
```

Comment:
This gives me pause. From TS docs:

> The @enum tag allows you to create an object literal whose members are all of a specified type. Unlike most object literals in Javascript, it does not allow other members.

The docs go on to say, later:

> Note that @enum is quite different from, and much simpler than, Typescript’s enum. 

This concerns me because A) it doesn't match what my own brain thinks _enum_ means and B) will it cause confusion because it is different than what TS itself thinks _enum_ means? It's a mild concern.

I want to make sure I understand the intent. Is this kind of like `{Object.<string,number>}` but you don't want to assert that the keys are a particular type?

Reply:
Honestly, I didn't think about this too deeply and will need to learn more myself. `LinkType` type is logically an enum and using `@enum` made TS happy.